### PR TITLE
Make Linting Consistent Again (Or Something)

### PR DESCRIPTION
**Pull request by jamiebrynes7**
#### Description
This PR aims to make linting more consistent by specifying line-lengths in the `.editorconfig` files. Also moved the `.editorconfig` to the root directory which will be inherited by all sub-directories. 

#### Tests
N/A
#### Documentation
N/A
#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.

### CODE REVIEW

File: code_generator/.editorconfig [hunk 1]
```diff
@@ -10,6 +10,10 @@ charset = utf-8
 insert_final_newline = true
```
Comment (jared-improbable) [inline code_generator/.editorconfig]: Any reason this and the config in `tools` are separate? Can we unify everything?

Comment (jamiebrynes7) [reply code_generator/.editorconfig]: I don't think so unless we add soft linking to the setup scripts, which we could do but feels a little brittle. 

Its not ideal to have it replicated though

Comment (jared-improbable) [reply code_generator/.editorconfig]: Why can't the single file live in the root of the GDK folder?

Comment (jamiebrynes7) [reply code_generator/.editorconfig]: Oh I misread the docs 🤦‍♂️ We can just have a parent one. Will change

File: tools/DocsLinter/SimplifiedMarkdownDoc.cs [hunk 2]
```diff
@@ -6,176 +6,176 @@
...
-  ///   A data class that represents a local link. I.e. - "../README.md#heading"
-  /// </summary>
-  public struct LocalLink : ILink
-  {
-    public Heading? Heading;
-    public string FilePath;
 
     /// <summary>
-    ///   Constructor for that parses the Markdig link object.
+    ///     Abstract base class for a link.
```
Comment (firtina-improbable) [inline tools/DocsLinter/SimplifiedMarkdownDoc.cs]: s/base class/interface/

File: workers/unity/Assets/Playground/Scripts/Player/ProcessLaunchCommandSystem.cs [hunk 3]
```diff
@@ -18,7 +18,8 @@ private struct LaunchCommandData
             [ReadOnly] public EntityArray Entity;
             public ComponentDataArray<SpatialOSLauncher> Launcher;
 
-            [ReadOnly] public ComponentArray<CommandRequests<Generated.Playground.Launcher.LaunchEntity.Request>> CommandRequests;
```
Comment (firtina-improbable) [inline workers/unity/Assets/Playground/Scripts/Player/ProcessLaunchCommandSystem.cs]: Is this PR going to fix this file flip flopping

Comment (jamiebrynes7) [reply workers/unity/Assets/Playground/Scripts/Player/ProcessLaunchCommandSystem.cs]: Yes it should by explicitly setting the line-length.

File: tools/DocsLinter/SimplifiedMarkdownDoc.cs (hunk 2 above)
Comment (jamiebrynes7) [reply tools/DocsLinter/SimplifiedMarkdownDoc.cs]: Driveby time 🔫

File: workers/unity/Assets/Gdk/Core/Components/ISpatialComponentData.cs [hunk 4]
```diff
@@ -1,3 +1,5 @@
+using Unity.Mathematics;
```
Comment (jessicafalk) [inline workers/unity/Assets/Gdk/Core/Components/ISpatialComponentData.cs]: remove this line. We don't need it anymore :)

**Pull request merged**